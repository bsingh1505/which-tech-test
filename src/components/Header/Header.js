function Header() {
  return (
    <div className="App">
      <header className="App-header">
        <p data-testid="header-content">Practise multiplications at your own pace and have fun!!!</p>
      </header>
    </div>
  );
}

export default Header;
