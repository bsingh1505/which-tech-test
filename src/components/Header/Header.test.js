import { render } from "@testing-library/react";
import Header from "./Header";

const setup = (overrideProps) => {
  const props = {
    multiplicationsUpto: 10,
    ...overrideProps,
  };
  const { container, getByTestId } = render(<Header {...props} />);
  return {
    container,
    props,
    getByTestId,
  };
};

describe("Header component", () => {
  test(":: should render component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  test(":: check the content", () => {
    const { getByTestId } = setup();
    const header = getByTestId("header-content");
    expect(header).toHaveTextContent("Practise multiplications at your own pace and have fun!!!");
  });
});
