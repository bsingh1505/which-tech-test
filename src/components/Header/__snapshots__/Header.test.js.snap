// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component :: should render component 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <p
        data-testid="header-content"
      >
        Practise multiplications at your own pace and have fun!!!
      </p>
    </header>
  </div>
</div>
`;
